### DISCUSSION TITLE
Why isn't this light and asset in Spoke casting a shadow?

### DISCUSSION THREAD

**Discussion by SeanBannister** [mozilla/hubs#2708] [General Questions & Discussion] [answered]
I'm new to lighting 3D scenes (thanks Dom and Jim for last nights [live stream](https://www.twitch.tv/videos/688663777?filter=archives&sort=time)), it's raising a few questions. For example the following scene ([available here](https://hubs.mozilla.com/scenes/qBpGWa8/light-test)) I've placed a "point light" (also tested with directional light) above an asset that's set to cast a shadow on to an asset set to receive a shadow but I'm seeing no shadow. I was expecting a shadow on the ground plane.

![image](https://user-images.githubusercontent.com/197231/88443689-a80da100-ce5c-11ea-8d9f-56726d16ceeb.png)

**Comment by cvanmeurs** [comment]
I was able to get shadows to appear in the preview image when publishing to Hubs.  I don't believe they are going to show up in Spoke because they need to bake to a texture when you publish the scene.  However I'm not seeing them when I get to the Hubs scene - that seems like a bug.  I'll see if I can find out.


![lightTest](https://user-images.githubusercontent.com/63256260/88446335-08113280-cdde-11ea-80ab-136eda6868cc.png)

**Comment by j-conrad** [comment]
Check the upper right of the 3d window. There should be a drop down menu where you can change it from ‘Lit’ to showing shadows.

**Reply by SeanBannister** [reply]
Just trying to get my head around this... when I click the dropdown and select "shadows" it doesn't show the shadow in Spoke but does when I publish to Hubs. But every time you load a scene in Spoke it resets that dropdown back to "lit", so I have to ensure I change it back to "shadows" every time before I publish if I want shadows?

I might be missing something, is this right/bug/or something that needs improving?

**Comment by robertlong** [comment, marked as answer]
This is likely due to an outstanding bug in Spoke: https://github.com/mozilla/Spoke/issues/861 We can continue tracking this issue there.

This would be fixed if we could update to the most recent version of three.js. We use a custom fork of three.js for Hubs and Spoke. We can't currently update that fork because three.js dropped support for WebVR and Firefox desktop does not yet have support for WebXR.

**Reply by SeanBannister** [reply]
Thanks Robert appreciate the response!
